src/tri/_tri.cpp

```diff
@@ -650,24 +645,26 @@ PyObject* TriContourGenerator::contour_line_to_segs_and_kinds(const Contour& con
             contour_line.front() == contour_line.back())
             *(codes_ptr-1) = CLOSEPOLY;
 
-        if (PyList_SetItem(vertices_list, i, (PyObject*)segs) ||
-            PyList_SetItem(codes_list, i, (PyObject*)codes)) {
-            Py_XDECREF(segs);
-            Py_XDECREF(codes);
-            PyErr_SetString(PyExc_RuntimeError,
-                            "Unable to set contour segments and kind codes");
-            return NULL;
+        if (PyList_SetItem(vertices_list, i, segs.pyobj()) ||
```

Comment:
Should this be `segs.pyobj()` or `segs.pyobj_steal()` as used elsewhere? I have a vague recollection of getting this wrong before when dealing with setting/appending list items, so this is probably correct.

Reply:
`PyList_SetItem` steals a ref, so this must be `.pyobj`.

Reply:
Thanks for checking this.